src/features/core/cookieConsent/ketch/browser.js

```diff
@@ -0,0 +1,87 @@
+import logger from '../../../../utils/logUtil';
+
+/* eslint-disable class-methods-use-this */
+class Ketch {
+  constructor() {
+    this.userConsentedPurposes = [];
+    this.userDeniedPurposes = [];
+
+    // updateKetchConsent callback function to update current consent purpose state
+    // this will be called from ketch rudderstack plugin
+    window.updateKetchConsent = (consent) => {
+      if (consent.purposes) {
+        const consentState = consent.purposes;
+        this.userConsentedPurposes = [];
+        this.userDeniedPurposes = [];
+        Object.entries(consentState).forEach((e) => {
+          const purposeCode = e[0];
+          const isConsented = e[1];
+          if (isConsented) {
+            this.userConsentedPurposes.push(purposeCode);
+          } else {
+            this.userDeniedPurposes.push(purposeCode);
+          }
+        });
+      }
+    };
+
+    // getKetchUserConsentedPurposes returns current ketch opted-in purposes
+    window.getKetchUserConsentedPurposes = () => this.userConsentedPurposes.slice();
```

Comment:
Why do we need `window.getKetchUserConsentedPurposes` or `window.getKetchUserDeniedPurposes`?